tests/DCPS/optional/README.rst

```diff
@@ -0,0 +1,10 @@
+###############
```

Comment:
This needs to be updated with an actual description.

Reply:
It was previously mentioned that it might be better to move these tests to the xcdr test, but can't directly as it doesn't require C++17. If the tests don't need to be moved in this PR I can add a description and maybe a note to  move the tests another time